great_expectations/rule_based_profiler/rule_based_profiler.py

```diff
@@ -815,6 +822,59 @@ def run_profiler(
 
         return result
 
+    @staticmethod
+    def run_profiler_on_data(
+        data_context: "DataContext",  # noqa: F821
+        profiler_store: ProfilerStore,
+        batch_request: Union[dict, BatchRequest, RuntimeBatchRequest],
+        name: Optional[str] = None,
+        ge_cloud_id: Optional[str] = None,
+        expectation_suite_name: Optional[str] = None,
+        include_citation: bool = True,
+    ) -> ExpectationSuite:
+        profiler: RuleBasedProfiler = RuleBasedProfiler.get_profiler(
+            data_context=data_context,
+            profiler_store=profiler_store,
+            name=name,
+            ge_cloud_id=ge_cloud_id,
+        )
+
+        rules: Dict[
+            str, Dict[str, Any]
+        ] = profiler._generate_rule_overrides_from_batch_request(batch_request)
+
+        result: ExpectationSuite = profiler.run(
+            rules=rules,
+            expectation_suite_name=expectation_suite_name,
+            include_citation=include_citation,
+        )
+        return result
+
+    def _generate_rule_overrides_from_batch_request(
+        self, batch_request: Union[dict, BatchRequest, RuntimeBatchRequest]
+    ) -> Dict[str, Dict[str, Any]]:
+        rules: List[Rule] = self.rules
+        if not isinstance(batch_request, dict):
+            batch_request = get_batch_request_as_dict(batch_request)
+            logger.info("Converted batch request to dictionary: %s", batch_request)
+
+        resulting_rules: Dict[str, Dict[str, Any]] = {}
+
+        for rule in rules:
+            domain_builder = rule.domain_builder
+            if isinstance(domain_builder, ColumnDomainBuilder):
+                domain_builder.batch_request = copy.deepcopy(batch_request)
+                domain_builder.batch_request["data_connector_query"] = {"index": -1}
+
+            parameter_builders = rule.parameter_builders
```

Comment:
@cdkini Not required, but might be interesting to add a comment that we are passing *ALL* batches to the parameter builder.  If not used carefully, a bias might creep in to the resulting estimates computed by the parameter builders.  So if the user crafts the batch request carefully, where there either is no notion of "current/active" batch, or it is excluded, then everything should be fine.